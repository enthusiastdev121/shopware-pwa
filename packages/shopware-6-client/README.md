# API client for Shopware 6

Compatibility with Shopware 6:  
[![shopware-ver](https://img.shields.io/badge/version-6.1.1-orange)](https://github.com/shopware/platform/releases/tag/v6.1.1)

## Installing

Using yarn:

```bash
$ yarn add @shopware-pwa/shopware-6-client
```


## Usage example

A simple example of how to use API client to connect with your Shopware instance.

In main project file setup connection informations.
```js
import {setup} from "@shopware-pwa/shopware-6-client"

setup({
  endpoint: 'https://address-to-my-shopware-instance.com',
  accessToken: 'myaccesstoken'
})
```

And then anywhere in your project, you can use API Client methods:

```js
import { getCategories } from "@shopware-pwa/shopware-6-client"

// later in component

async mounted() {
  this.categories = await getCategories();
}
```