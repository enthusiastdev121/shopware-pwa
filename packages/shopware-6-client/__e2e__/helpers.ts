/**
 * Changes deeply specific property. Can be used in snapshot tests to mock changing value.
 */
export function deepChangeProperty(
  obj: any,
  property: string,
  value: any = "mockedValue"
) {
  deepChangeProperties(obj, [property, value]);
}

export function escapeQuotes(value: string) {
  return value.split('"').join("'");
}

export function deepEscapeProperties(obj: any, properties: string[]) {
  for (var prop in obj) {
    if (properties.includes(prop)) {
      if (typeof obj[prop] == "string") {
        obj[prop] = escapeQuotes(obj[prop]);
      }
    } else if (obj[prop] && obj[prop] === Object(obj[prop])) {
      deepEscapeProperties(obj[prop], properties);
    }
  }
}

export function deepChangeProperties(
  obj: any,
  properties: string[],
  value: any = "mockedValue"
) {
  for (var prop in obj) {
    if (properties.includes(prop) && Array.isArray(obj[prop])) {
      obj[prop] = [value];
    } else if (properties.includes(prop)) {
      obj[prop] = value;
    } else if (obj[prop] && obj[prop] === Object(obj[prop])) {
      deepChangeProperties(obj[prop], properties, value);
    }
  }
}
