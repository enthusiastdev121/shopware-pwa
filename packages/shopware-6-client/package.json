{
  "name": "@shopware-pwa/shopware-6-client",
  "version": "0.1.0-alpha.7",
  "description": "Rest API client for Shopware 6.",
  "main": "dist/shopware-6-client.cjs.js",
  "module": "dist/shopware-6-client.esm-bundler.js",
  "files": [
    "dist"
  ],
  "types": "dist/shopware-6-client.d.ts",
  "unpkg": "dist/shopware-6-client.global.js",
  "sideEffects": false,
  "repository": {
    "type": "git",
    "url": "git+https://github.com/DivanteLtd/shopware-pwa"
  },
  "buildOptions": {
    "name": "Shopware6Client",
    "formats": [
      "esm-bundler",
      "cjs"
    ]
  },
  "peerDependencies": {
    "axios": "^0.19.2",
    "https": "^1.0.0"
  },
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  }
}
