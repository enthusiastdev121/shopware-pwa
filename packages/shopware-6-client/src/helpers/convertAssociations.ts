import {
  Association,
  ShopwareAssociation,
} from "@shopware-pwa/commons/interfaces/search/Association";

export function convertAssociations(
  associations: Association[] = []
): ShopwareAssociation | undefined {
  if (!Array.isArray(associations) || !associations.length) return;
  let shopwareAssociations: ShopwareAssociation = {};
  associations.forEach((association) => {
    shopwareAssociations[association.name] = association.associations
      ? {
          associations: convertAssociations(association.associations),
        }
      : {};
  });
  return shopwareAssociations;
}
