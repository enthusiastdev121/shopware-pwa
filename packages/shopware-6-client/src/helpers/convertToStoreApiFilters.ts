import {
  SearchFilter,
  SearchFilterType,
} from "@shopware-pwa/commons/interfaces/search/SearchFilter";

const concatIds = (ids: string[]) => ids.join("|");
const isFilterForProperty = (property: string, filter: any) =>
  filter.hasOwnProperty("field") && filter.field === property;

interface StoreApiListingFilters {
  manufacturer?: string;
  properties?: string;
  rating?: number; // to be handled later on
  "shipping-free"?: boolean; // to be handled later on
  "min-price"?: number; // to be handled later on
  "max-price"?: number; // to be handled later on
}

interface GenericFilter extends SearchFilter {
  value?: string[];
  queries?: GenericFilter[];
}

export function convertToStoreApiFilters(
  filters: Array<GenericFilter>
): StoreApiListingFilters {
  let params: StoreApiListingFilters = {};

  if (!filters || !filters.length) {
    return {};
  }

  for (const filter of filters) {
    if (isFilterForProperty("manufacturerId", filter) && filter.value) {
      params.manufacturer = concatIds(filter.value);
    }
    if (filter.type === SearchFilterType.MULTI && filter.queries) {
      for (const subFilter of filter.queries) {
        /* istanbul ignore else */
        if (isFilterForProperty("propertyIds", subFilter)) {
          params.properties = concatIds(subFilter.value as any);
        }
      }
    }
  }

  return params;
}
