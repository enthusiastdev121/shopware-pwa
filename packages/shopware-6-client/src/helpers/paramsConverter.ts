import { SearchCriteria } from "../interfaces/search/SearchCriteria";

export interface Params {
  page?: number;
  limit?: number;
  sort?: string;
  filter?: any;
}

/**
 * @description Combines parameters into one object
 */
export const getParams = (searchCriteria?: SearchCriteria): Params | null => {
  let params = {};

  if (!searchCriteria) return params;

  if (searchCriteria.pagination) {
    params = Object.assign(params, searchCriteria.pagination);
  }

  // if (sort) {
  //   params = Object.assign(params, sort);
  // }

  if (searchCriteria.filters) {
    params = Object.assign(params, searchCriteria.filters);
  }

  return params;
};
