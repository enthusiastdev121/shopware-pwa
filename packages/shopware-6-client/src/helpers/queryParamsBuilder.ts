import queryString from "query-string";

const ARRAY_FORMAT = "separator";
const ARRAY_FORMAT_SEPARATOR = "|";
const SKIP_NULL = true;
const SORT = false;

export const getQueryString = (params: any) =>
  typeof params === "string"
    ? params
    : queryString.stringify(params, {
        arrayFormat: ARRAY_FORMAT,
        arrayFormatSeparator: ARRAY_FORMAT_SEPARATOR,
        skipNull: SKIP_NULL,
        sort: SORT,
      });
