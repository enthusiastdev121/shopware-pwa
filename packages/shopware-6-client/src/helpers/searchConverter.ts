import { SearchCriteria } from "@shopware-pwa/commons/interfaces/search/SearchCriteria";
import {
  NotFilter,
  MultiFilter,
  RangeFilter,
  EqualsFilter,
  EqualsAnyFilter,
  SearchFilterType,
} from "@shopware-pwa/commons/interfaces/search/SearchFilter";
import { convertAssociations } from "./convertAssociations";
import { PaginationLimit } from "@shopware-pwa/commons/interfaces/search/Pagination";
import { config } from "@shopware-pwa/shopware-6-client";
import { ShopwareAssociation } from "@shopware-pwa/commons/interfaces/search/Association";
import { Grouping } from "@shopware-pwa/commons/interfaces/search/Grouping";

/**
 * @alpha
 */
export interface ShopwareParams {
  page?: number;
  limit?: number;
  sort?: string;
  term?: string;
  filter?: (
    | NotFilter
    | MultiFilter
    | EqualsFilter
    | EqualsAnyFilter
    | RangeFilter
  )[];
  associations?: ShopwareAssociation;
  grouping?: Grouping;
}

export const convertSearchCriteria = (
  searchCriteria?: SearchCriteria
): ShopwareParams => {
  let params: ShopwareParams = {};

  if (!searchCriteria) return params;
  const { filters, sort, pagination, configuration, term } = searchCriteria;

  if (pagination) {
    const { limit, page } = pagination;
    if (limit && Object.values(PaginationLimit).includes(limit))
      params.limit = limit;
    if (page) {
      params.page = page;
      if (!params.limit) params.limit = config.defaultPaginationLimit;
    }
  }

  if (sort) {
    let prefix = sort.desc ? "-" : "";
    params.sort = `${prefix}${sort.field}`;
  }

  if (filters && filters.length) {
    params.filter = filters;
  }

  if (configuration && configuration.associations) {
    params.associations = convertAssociations(configuration.associations);
  }

  if (configuration?.grouping) {
    params.grouping = configuration.grouping;
  }

  // fulltext query (works for every entity so can be global)
  if (term) {
    params.term = term;
  }

  // add extra grouping option and additional filter to prevent displaying redundand products.
  if (!configuration || (configuration && !configuration.displayParents)) {
    if (!Array.isArray(params.filter)) {
      params.filter = [];
    }
    params.grouping = {
      // prevent displaying parent instances of the product
      field: "displayGroup",
    };
    params.filter.push({
      type: SearchFilterType.NOT,
      queries: [
        {
          type: SearchFilterType.EQUALS,
          field: "displayGroup",
          value: null,
        },
      ],
    });
  }

  return params;
};
