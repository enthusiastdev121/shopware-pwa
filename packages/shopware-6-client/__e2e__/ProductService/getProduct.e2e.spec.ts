import { getProduct } from "@shopware-pwa/shopware-6-client";
import { deepChangeProperty } from "../helpers";

describe("shopware-6-client - E2E - ProductService - getProduct", () => {
  it("should fetch real product", async () => {
    const result = await getProduct("044a190a54ab4f06803909c3ee8063ef");
    deepChangeProperty(result, "categoryTree");
    deepChangeProperty(result, "listingPrices");
    expect(result).toMatchSnapshot();
  });

  it("should return error on not existing product", async () => {
    try {
      await getProduct("proooduct");
      expect("didn't throw an error").toEqual("should throw an error");
    } catch (e) {
      expect(e).toMatchSnapshot();
    }
  });
});
