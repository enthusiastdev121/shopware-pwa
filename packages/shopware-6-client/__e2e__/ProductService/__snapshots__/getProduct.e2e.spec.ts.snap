// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shopware-6-client - E2E - ProductService - getProduct should fetch real product 1`] = `
Object {
  "_uniqueIdentifier": "044a190a54ab4f06803909c3ee8063ef",
  "active": true,
  "autoIncrement": 59,
  "available": true,
  "availableStock": 37,
  "calculatedListingPrice": Object {
    "extensions": Array [],
    "from": Object {
      "calculatedTaxes": Array [
        Object {
          "extensions": Array [],
          "price": 490.00000000000006,
          "tax": 78.24,
          "taxRate": 19,
        },
      ],
      "extensions": Array [],
      "quantity": 1,
      "referencePrice": null,
      "taxRules": Array [
        Object {
          "extensions": Array [],
          "percentage": 100,
          "taxRate": 19,
        },
      ],
      "totalPrice": 490,
      "unitPrice": 490,
    },
    "to": Object {
      "calculatedTaxes": Array [
        Object {
          "extensions": Array [],
          "price": 886,
          "tax": 141.46,
          "taxRate": 19,
        },
      ],
      "extensions": Array [],
      "quantity": 1,
      "referencePrice": null,
      "taxRules": Array [
        Object {
          "extensions": Array [],
          "percentage": 100,
          "taxRate": 19,
        },
      ],
      "totalPrice": 886,
      "unitPrice": 886,
    },
  },
  "calculatedPrice": Object {
    "calculatedTaxes": Array [
      Object {
        "extensions": Array [],
        "price": 48,
        "tax": 7.66,
        "taxRate": 19,
      },
    ],
    "extensions": Array [],
    "quantity": 1,
    "referencePrice": null,
    "taxRules": Array [
      Object {
        "extensions": Array [],
        "percentage": 100,
        "taxRate": 19,
      },
    ],
    "totalPrice": 48,
    "unitPrice": 48,
  },
  "calculatedPrices": Array [
    Object {
      "calculatedTaxes": Array [
        Object {
          "extensions": Array [],
          "price": 8860,
          "tax": 1414.62,
          "taxRate": 19,
        },
      ],
      "extensions": Array [],
      "quantity": 10,
      "referencePrice": null,
      "taxRules": Array [
        Object {
          "extensions": Array [],
          "percentage": 100,
          "taxRate": 19,
        },
      ],
      "totalPrice": 8860,
      "unitPrice": 886,
    },
    Object {
      "calculatedTaxes": Array [
        Object {
          "extensions": Array [],
          "price": 5390.000000000001,
          "tax": 860.59,
          "taxRate": 19,
        },
      ],
      "extensions": Array [],
      "quantity": 11,
      "referencePrice": null,
      "taxRules": Array [
        Object {
          "extensions": Array [],
          "percentage": 100,
          "taxRate": 19,
        },
      ],
      "totalPrice": 5390,
      "unitPrice": 490,
    },
  ],
  "categories": null,
  "categoriesRo": null,
  "categoryTree": Array [
    "mockedValue",
  ],
  "childCount": 6,
  "children": null,
  "cover": Object {
    "_uniqueIdentifier": "1c739fdf41f94aedb87262b6a03f8886",
    "createdAt": "2019-12-05T13:52:24+00:00",
    "customFields": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
      "internal_mapping_storage": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "id": "1c739fdf41f94aedb87262b6a03f8886",
    "media": Object {
      "_uniqueIdentifier": "6b056c5a7c0c4e68933aa44e806e387e",
      "alt": null,
      "createdAt": "2019-12-05T13:52:22+00:00",
      "customFields": null,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
        "internal_mapping_storage": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "fileExtension": "png",
      "fileName": "fashion_ZA027049_008_t4_6.png",
      "fileSize": 534698,
      "hasFile": true,
      "id": "6b056c5a7c0c4e68933aa44e806e387e",
      "metaData": Object {
        "height": 906,
        "type": 3,
        "width": 600,
      },
      "mimeType": "image/png",
      "private": false,
      "tags": null,
      "thumbnails": Array [],
      "title": null,
      "translated": Object {
        "alt": null,
        "customFields": Array [],
        "title": null,
      },
      "translations": null,
      "updatedAt": "2019-12-05T13:52:24+00:00",
      "uploadedAt": "2019-12-05T13:52:24+00:00",
      "url": "https://shopware6-demo.vuestorefront.io/media/0b/5f/78/1575553944/fashion_ZA027049_008_t4_6.png.png",
      "versionId": null,
    },
    "mediaId": "6b056c5a7c0c4e68933aa44e806e387e",
    "position": 1,
    "product": null,
    "productId": "044a190a54ab4f06803909c3ee8063ef",
    "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
    "translated": Array [],
    "updatedAt": null,
    "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
  },
  "coverId": "1c739fdf41f94aedb87262b6a03f8886",
  "createdAt": "2019-10-08T09:42:16+00:00",
  "crossSellings": null,
  "customFields": null,
  "deliveryTime": null,
  "deliveryTimeId": null,
  "description": "Et magnam et earum eos et velit repellendus. Saepe vitae possimus impedit. Ut eligendi neque enim ducimus accusantium vel quis. Ut cupiditate recusandae deserunt ut et unde.",
  "displayGroup": null,
  "ean": null,
  "extensions": Object {
    "foreignKeys": Object {
      "_uniqueIdentifier": null,
      "createdAt": null,
      "extensions": Array [],
      "translated": Array [],
      "updatedAt": null,
      "versionId": null,
    },
  },
  "height": null,
  "id": "044a190a54ab4f06803909c3ee8063ef",
  "isCloseout": false,
  "isNew": false,
  "keywords": null,
  "length": null,
  "listingPrices": Array [
    "mockedValue",
  ],
  "mainCategories": null,
  "manufacturer": null,
  "manufacturerId": "79eb793025ee4fa3b628f9814bd7b010",
  "manufacturerNumber": null,
  "markAsTopseller": null,
  "maxPurchase": null,
  "media": null,
  "metaDescription": null,
  "metaTitle": null,
  "minPurchase": 1,
  "name": "Aerodynamic Iron Jetsilk Hallo",
  "optionIds": null,
  "options": null,
  "packUnit": null,
  "parent": null,
  "parentId": null,
  "parentVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
  "price": Array [
    Object {
      "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
      "extensions": Array [],
      "gross": 48,
      "linked": true,
      "net": 40.33613445378151,
    },
  ],
  "prices": Array [
    Object {
      "_uniqueIdentifier": "07afda1ecf3244f18b167ab8529c9c1d",
      "createdAt": "2019-10-08T09:42:16+00:00",
      "customFields": null,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "id": "07afda1ecf3244f18b167ab8529c9c1d",
      "price": Array [
        Object {
          "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
          "extensions": Array [],
          "gross": 245,
          "linked": true,
          "net": 205.8823529411765,
        },
      ],
      "product": null,
      "productId": "044a190a54ab4f06803909c3ee8063ef",
      "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "quantityEnd": null,
      "quantityStart": 11,
      "ruleId": "3641e978b8b9478f8865d3feed03f4ca",
      "translated": Array [],
      "updatedAt": null,
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
    },
    Object {
      "_uniqueIdentifier": "1c90794d216d4c3da641015737af1490",
      "createdAt": "2019-10-08T09:42:16+00:00",
      "customFields": null,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "id": "1c90794d216d4c3da641015737af1490",
      "price": Array [
        Object {
          "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
          "extensions": Array [],
          "gross": 490,
          "linked": true,
          "net": 411.764705882353,
        },
      ],
      "product": null,
      "productId": "044a190a54ab4f06803909c3ee8063ef",
      "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "quantityEnd": null,
      "quantityStart": 11,
      "ruleId": "51fba434c8004cc89bb70897167641a8",
      "translated": Array [],
      "updatedAt": null,
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
    },
    Object {
      "_uniqueIdentifier": "673b0a7456a64af2b1e6a7961ac6b7ef",
      "createdAt": "2019-10-08T09:42:16+00:00",
      "customFields": null,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "id": "673b0a7456a64af2b1e6a7961ac6b7ef",
      "price": Array [
        Object {
          "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
          "extensions": Array [],
          "gross": 886,
          "linked": true,
          "net": 744.5378151260504,
        },
      ],
      "product": null,
      "productId": "044a190a54ab4f06803909c3ee8063ef",
      "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "quantityEnd": 10,
      "quantityStart": 1,
      "ruleId": "51fba434c8004cc89bb70897167641a8",
      "translated": Array [],
      "updatedAt": null,
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
    },
    Object {
      "_uniqueIdentifier": "eed481dd9ed3456db0c550aa6731a7ec",
      "createdAt": "2019-10-08T09:42:16+00:00",
      "customFields": null,
      "extensions": Object {
        "foreignKeys": Object {
          "_uniqueIdentifier": null,
          "createdAt": null,
          "extensions": Array [],
          "translated": Array [],
          "updatedAt": null,
          "versionId": null,
        },
      },
      "id": "eed481dd9ed3456db0c550aa6731a7ec",
      "price": Array [
        Object {
          "currencyId": "b7d2554b0ce847cd82f3ac9bd1c0dfca",
          "extensions": Array [],
          "gross": 823,
          "linked": true,
          "net": 691.5966386554622,
        },
      ],
      "product": null,
      "productId": "044a190a54ab4f06803909c3ee8063ef",
      "productVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
      "quantityEnd": 10,
      "quantityStart": 1,
      "ruleId": "3641e978b8b9478f8865d3feed03f4ca",
      "translated": Array [],
      "updatedAt": null,
      "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
    },
  ],
  "productManufacturerVersionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
  "productMediaVersionId": null,
  "productNumber": "eea0f69ec02d44f7a4224272b3d99478",
  "productReviews": null,
  "properties": null,
  "propertyIds": Array [
    "003a81c935674e90a48eb0a52a6575dc",
    "0a4ff74816154e7fa1ba043d35b46014",
    "1605f67618b2421ab565fff5da8686d0",
    "1d6602dbd8254541b42272f8287860aa",
    "3647feeb79fd43b0b371e8392e74ba5b",
    "390e557fe42e46d7a0651c375b1757de",
    "43231c2e5de1434a8bda2ddd0cd3239c",
    "5ce324943a694a1d833dcc6b798f1a47",
    "872a36ad229f4a2bb747e8f504bd8d67",
    "a4896fec5a3e496f8f8d5f50e91cc09d",
    "b1a7e209d0274ad2a1b47beed2adf05e",
    "c8c382b1e90748d2bb5719f456dc2cb9",
    "c9f2fac6f27043d8ad6ae98dd95ccd91",
    "ce0d7c8cccae452aaa5e78265c28dccb",
    "d9231ebabf5e4e5e8695e5e5f5c9d717",
    "e29259096632459fbb5f5af70675bb98",
    "e3255d80f51e44bba63c9345555fe629",
    "fbc9806c95394b6d9def90ca80b9515b",
  ],
  "purchasePrice": null,
  "purchaseSteps": 1,
  "purchaseUnit": null,
  "ratingAverage": null,
  "referenceUnit": null,
  "releaseDate": null,
  "restockTime": 3,
  "seoUrls": null,
  "shippingFree": false,
  "sortedProperties": null,
  "stock": 37,
  "tagIds": null,
  "tags": null,
  "tax": Object {
    "_uniqueIdentifier": "422551e3c12046799dd3640887c7d249",
    "createdAt": "2019-10-08T09:41:16+00:00",
    "customFields": null,
    "extensions": Object {
      "foreignKeys": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
      "internal_mapping_storage": Object {
        "_uniqueIdentifier": null,
        "createdAt": null,
        "extensions": Array [],
        "translated": Array [],
        "updatedAt": null,
        "versionId": null,
      },
    },
    "id": "422551e3c12046799dd3640887c7d249",
    "name": "19%",
    "products": null,
    "taxRate": 19,
    "translated": Array [],
    "updatedAt": null,
    "versionId": null,
  },
  "taxId": "422551e3c12046799dd3640887c7d249",
  "translated": Object {
    "customFields": Array [],
    "description": "Et magnam et earum eos et velit repellendus. Saepe vitae possimus impedit. Ut eligendi neque enim ducimus accusantium vel quis. Ut cupiditate recusandae deserunt ut et unde.",
    "keywords": null,
    "metaDescription": null,
    "metaTitle": null,
    "name": "Aerodynamic Iron Jetsilk Hallo",
    "packUnit": null,
  },
  "translations": null,
  "unit": null,
  "unitId": null,
  "updatedAt": "2019-12-05T13:52:25+00:00",
  "versionId": "0fa91ce3e96a4bc2be4bd9ce752c3425",
  "weight": null,
  "width": null,
}
`;

exports[`shopware-6-client - E2E - ProductService - getProduct should return error on not existing product 1`] = `[Error: Request failed with status code 400]`;
